From d955add31b6f1985f526a6bf24a81ecf2120773b Mon Sep 17 00:00:00 2001
From: napari-bot <81196843+napari-bot@users.noreply.github.com>
Date: Thu, 22 Jun 2023 02:23:08 +0200
Subject: [PATCH] test: [Automatic] Constraints upgrades: `dask`, `fsspec`,
 `hypothesis`, `imageio`, `ipython`, `napari-plugin-manager`, `napari-svg`,
 `numpy`, `psygnal`, `pydantic`, `pyqt6`, `pytest`, `rich`, `scikit-image`,
 `virtualenv`, `zarr` (#5963)

This PR is automatically created and updated by this napari GitHub
Action:

https://github.com/napari/napari/tree/main/.github/workflows/upgrade_test_contraints.yml
It ensures that dependencies are tested with the most recent version.

The updated packages are:

 * cmake
 * contourpy
 * coverage
 * dask
 * filelock
 * fonttools
 * fsspec
 * hypothesis
 * imageio
 * importlib-metadata
 * ipykernel
 * ipython
 * jupyter-core
 * lit
 * markdown-it-py
 * markupsafe
 * napari-plugin-manager
 * napari-svg
 * numpy
 * pip
 * platformdirs
 * pluggy
 * psygnal
 * pydantic
 * pyparsing
 * pyqt6
 * pytest
 * rich
 * scikit-image
 * setuptools
 * typing-extensions
 * urllib3
 * virtualenv
 * zarr

Co-authored-by: napari-bot <napari-bot@users.noreply.github.com>
---
 .../workflows/upgrade_test_constraints.yml    |  44 +-
 .../humanize/src/humanize/__init__.py         |   9 +-
 napari/utils/_tests/test_misc.py              |  11 +
 resources/constraints/constraints_py3.10.txt  | 116 ++--
 resources/constraints/constraints_py3.11.txt  | 101 +--
 resources/constraints/constraints_py3.8.txt   | 113 ++--
 resources/constraints/constraints_py3.9.txt   | 118 ++--
 .../constraints_py3.9_examples.txt            | 582 ++++++++++++++++++
 resources/constraints/version_denylist.txt    |   4 +-
 resources/requirements_mypy.txt               |  92 ++-
 setup.cfg                                     |   6 +-
 tools/check_updated_packages.py               |  52 +-
 tools/get_auto_upgrade_branch_name.py         |  60 ++
 13 files changed, 1075 insertions(+), 233 deletions(-)
 create mode 100644 tools/get_auto_upgrade_branch_name.py

diff --git a/.github/workflows/upgrade_test_constraints.yml b/.github/workflows/upgrade_test_constraints.yml
index 28177608..de199674 100644
--- a/.github/workflows/upgrade_test_constraints.yml
+++ b/.github/workflows/upgrade_test_constraints.yml
@@ -6,20 +6,31 @@ on:
     # Runs every Monday at 8:00 UTC (4:00 Eastern)
     - cron: '0 8 * * 1'
 
+  push:
+    paths:
+        - 'resources/constraints/version_denylist.txt'
+        - 'resources/constraints/version_denylist_examples.txt'
+        - 'resources/requirements_mypy.in'
+        - 'setup.cfg'
+        - '.github/workflows/upgrade_test_constraints.yml'
+        - 'tools/get_auto_upgrade_branch_name.py'
+        - 'tools/check_updated_packages.py'
+
+concurrency:
+  group: test-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   upgrade:
     name: Upgrade & Open Pull Request
     runs-on: ubuntu-latest
-    env:
-      # This branch will receive updates each time the workflow runs
-      # It doesn't matter if it's deleted when merged, it'll be re-created
-      BRANCH_NAME: auto-dependency-upgrades
     steps:
       - uses: actions/checkout@v3
         with:
           # [Optional] Use a separate key to automatically execute checks on the resulting PR
           # https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs
           ssh-key: ${{ secrets.DEPLOY_KEY }}
+          fetch-depth: 0
 
       # START PYTHON DEPENDENCIES
       - uses: actions/setup-python@v4
@@ -42,9 +53,18 @@ jobs:
           python-version: "3.11"
           cache: pip
           cache-dependency-path: 'setup.cfg'
+      - name: "Calculate branch name"
+        id: branch_name
+        run: |
+          python tools/get_auto_upgrade_branch_name.py --verbose
+          git fetch --all
+          git branch -a
+          echo "branch_name=$(python tools/get_auto_upgrade_branch_name.py)" >> $GITHUB_OUTPUT
+          echo "base_branch_name=$(python tools/get_auto_upgrade_branch_name.py --base_branch)" >> $GITHUB_OUTPUT
 
       - name: Upgrade Python dependencies
         # ADD YOUR CUSTOM DEPENDENCY UPGRADE COMMANDS BELOW
+        if: steps.branch_name.outputs.branch_name != 'skip'
         run: |
           flags=""
           # Explanation of below commands 
@@ -84,6 +104,7 @@ jobs:
       - name: Check updated packages
         id: packages
         run: |
+          git fetch --all
           python tools/check_updated_packages.py --main-packages
           python tools/check_updated_packages.py
           echo "main_packages=$(python tools/check_updated_packages.py --main-packages)" >> $GITHUB_OUTPUT
@@ -91,11 +112,22 @@ jobs:
           echo "$(python tools/check_updated_packages.py)" >> $GITHUB_OUTPUT
           echo "EOF" >> $GITHUB_OUTPUT
 
-      - name: Create PR updating vendored modules
+      - name: Create commit
+        if: (steps.branch_name.outputs.branch_name != 'skip') && (github.event_name == 'push')
+        run: |
+          git config --global user.name "napari-bot"
+          git config --global user.email "napari-bot@users.noreply.github.com"
+          git reset --soft `git merge-base origin/${{ steps.branch_name.outputs.base_branch_name }} HEAD`
+          git add -u
+          git diff-index --quiet HEAD -- || git commit -m "update constraints"
+          
+
+      - name: Create PR updating test constraints
         uses: peter-evans/create-pull-request@v5
         with:
           commit-message: Update bundle dependencies.
-          branch: ${{ env.BRANCH_NAME }}
+          branch: ${{ steps.branch_name.outputs.branch_name }}
+          base: ${{ steps.branch_name.outputs.base_branch_name }}
           delete-branch: true
           labels: maintenance
           title: "test: [Automatic] Constraints upgrades: ${{ steps.packages.outputs.main_packages }}"
diff --git a/napari/_vendor/experimental/humanize/src/humanize/__init__.py b/napari/_vendor/experimental/humanize/src/humanize/__init__.py
index 1b75509a..54265a2b 100644
--- a/napari/_vendor/experimental/humanize/src/humanize/__init__.py
+++ b/napari/_vendor/experimental/humanize/src/humanize/__init__.py
@@ -1,8 +1,7 @@
-import pkg_resources
-from napari._vendor.experimental.humanize.src.humanize.filesize import naturalsize
-from napari._vendor.experimental.humanize.src.humanize.i18n import activate, deactivate
-from napari._vendor.experimental.humanize.src.humanize.number import apnumber, fractional, intcomma, intword, ordinal, scientific
-from napari._vendor.experimental.humanize.src.humanize.time import (
+from .filesize import naturalsize
+from .i18n import activate, deactivate
+from .number import apnumber, fractional, intcomma, intword, ordinal, scientific
+from .time import (
     naturaldate,
     naturalday,
     naturaldelta,
diff --git a/napari/utils/_tests/test_misc.py b/napari/utils/_tests/test_misc.py
index f3406469..eb1a18ce 100644
--- a/napari/utils/_tests/test_misc.py
+++ b/napari/utils/_tests/test_misc.py
@@ -1,8 +1,10 @@
 from enum import auto
+from importlib.metadata import version as package_version
 from os.path import abspath, expanduser, sep
 from pathlib import Path
 
 import pytest
+from packaging.version import parse as parse_version
 
 from napari.utils.misc import (
     StringEnum,
@@ -198,6 +200,15 @@ def test_equality_operator():
         pick_equality_operator(xr.DataArray(np.ones((1, 1))))
         == _quiet_array_equal
     )
+
+
+@pytest.mark.skipif(
+    parse_version(package_version("numpy")) >= parse_version("1.25.0"),
+    reason="Numpy 1.25.0 return true for below comparison",
+)
+def test_equality_operator_silence():
+    import numpy as np
+
     eq = pick_equality_operator(np.asarray([]))
     # make sure this doesn't warn
     assert not eq(np.asarray([]), np.asarray([], '<U32'))
diff --git a/resources/constraints/constraints_py3.10.txt b/resources/constraints/constraints_py3.10.txt
index 18d390b3..2e0b30cc 100644
--- a/resources/constraints/constraints_py3.10.txt
+++ b/resources/constraints/constraints_py3.10.txt
@@ -30,7 +30,7 @@ build==0.10.0
     # via npe2
 cachey==0.2.1
     # via napari (setup.cfg)
-certifi==2022.12.7
+certifi==2023.5.7
     # via
     #   napari (setup.cfg)
     #   requests
@@ -42,17 +42,17 @@ click==8.1.3
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.3
+cmake==3.26.4
     # via triton
 comm==0.1.3
     # via ipykernel
-contourpy==1.0.7
+contourpy==1.1.0
     # via matplotlib
-coverage==7.2.3
+coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.4.0
+dask==2023.6.0
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
@@ -74,16 +74,16 @@ executing==1.2.0
     # via stack-data
 fasteners==0.18
     # via zarr
-filelock==3.12.0
+filelock==3.12.2
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.39.3
+fonttools==4.40.0
     # via matplotlib
-freetype-py==2.3.0
+freetype-py==2.4.0
     # via vispy
-fsspec==2023.4.0
+fsspec==2023.6.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -91,28 +91,28 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.74.0
+hypothesis==6.79.1
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.28.0
+imageio==2.31.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
-importlib-metadata==6.6.0
+importlib-metadata==6.7.0
     # via dask
 in-n-out==0.1.7
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.22.0
+ipykernel==6.23.2
     # via
     #   napari-console
     #   qtconsole
-ipython==8.12.0
+ipython==8.14.0
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -132,7 +132,7 @@ jupyter-client==8.2.0
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.0
+jupyter-core==5.3.1
     # via
     #   ipykernel
     #   jupyter-client
@@ -145,7 +145,7 @@ lazy-loader==0.2
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.2
+lit==16.0.6
     # via triton
 locket==1.0.0
     # via partd
@@ -153,9 +153,9 @@ lxml==4.9.2
     # via napari (setup.cfg)
 magicgui==0.7.2
     # via napari (setup.cfg)
-markdown-it-py==2.2.0
+markdown-it-py==3.0.0
     # via rich
-markupsafe==2.1.2
+markupsafe==2.1.3
     # via jinja2
 matplotlib==3.7.1
     # via napari (setup.cfg)
@@ -176,7 +176,9 @@ napari-plugin-engine==0.2.0
     #   napari (setup.cfg)
     #   napari-console
     #   napari-svg
-napari-svg==0.1.6
+napari-plugin-manager==0.1.0a2
+    # via napari (setup.cfg)
+napari-svg==0.1.10
     # via napari (setup.cfg)
 nest-asyncio==1.5.6
     # via ipykernel
@@ -185,10 +187,12 @@ networkx==3.1
     #   scikit-image
     #   torch
 npe2==0.7.0
-    # via napari (setup.cfg)
+    # via
+    #   napari (setup.cfg)
+    #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
-numpy==1.24.3
+numpy==1.25.0
     # via
     #   contourpy
     #   dask
@@ -248,7 +252,7 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.1 ; python_version >= "3.9"
+pandas==2.0.2 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   xarray
@@ -266,14 +270,14 @@ pillow==9.5.0
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.20.1
+pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.3.0
+platformdirs==3.7.0
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.0.0
+pluggy==1.2.0
     # via pytest
 pooch==1.7.0
     # via
@@ -285,7 +289,7 @@ psutil==5.9.5
     # via
     #   ipykernel
     #   napari (setup.cfg)
-psygnal==0.9.0
+psygnal==0.9.1
     # via
     #   app-model
     #   magicgui
@@ -295,7 +299,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.7
+pydantic==1.10.9
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -309,8 +313,10 @@ pygments==2.15.1
     #   sphinx
     #   superqt
 pyopengl==3.1.6
-    # via napari (setup.cfg)
-pyparsing==3.0.9
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+pyparsing==3.1.0
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
@@ -320,9 +326,9 @@ pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.1
     # via pyqt5
-pyqt6==6.5.0
+pyqt6==6.5.1
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.0
+pyqt6-qt6==6.5.1
     # via pyqt6
 pyqt6-sip==13.5.1
     # via pyqt6
@@ -340,7 +346,7 @@ pyside6-essentials==6.4.2
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.3.1
+pytest==7.3.2
     # via
     #   napari (setup.cfg)
     #   pytest-cov
@@ -348,13 +354,13 @@ pytest==7.3.1
     #   pytest-metadata
     #   pytest-pretty
     #   pytest-qt
-pytest-cov==4.0.0
+pytest-cov==4.1.0
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
 pytest-json-report==1.5.0
     # via -r resources/constraints/version_denylist.txt
-pytest-metadata==2.0.4
+pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
     # via napari (setup.cfg)
@@ -376,12 +382,12 @@ pyyaml==6.0
     #   dask
     #   napari (setup.cfg)
     #   npe2
-pyzmq==25.0.2
+pyzmq==25.1.0
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.1
+qtconsole==5.4.3
     # via
     #   napari (setup.cfg)
     #   napari-console
@@ -390,18 +396,19 @@ qtpy==2.3.1
     #   magicgui
     #   napari (setup.cfg)
     #   napari-console
+    #   napari-plugin-manager
     #   qtconsole
     #   superqt
-requests==2.28.2
+requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.3.4
+rich==13.4.2
     # via
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-scikit-image==0.20.0
+scikit-image==0.21.0
     # via napari (setup.cfg)
 scipy==1.10.1 ; python_version >= "3.9"
     # via
@@ -444,10 +451,13 @@ superqt==0.4.1
     # via
     #   magicgui
     #   napari (setup.cfg)
-sympy==1.11.1
+    #   napari-plugin-manager
+sympy==1.12
     # via torch
-tensorstore==0.1.36
-    # via napari (setup.cfg)
+tensorstore==0.1.37
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
 tifffile==2023.4.12
     # via
     #   napari (setup.cfg)
@@ -463,11 +473,11 @@ toolz==0.12.0
     #   dask
     #   napari (setup.cfg)
     #   partd
-torch==2.0.0
+torch==2.0.1
     # via
     #   napari (setup.cfg)
     #   triton
-tornado==6.3.1
+tornado==6.3.2
     # via
     #   ipykernel
     #   jupyter-client
@@ -484,22 +494,24 @@ traitlets==5.9.0
     #   qtconsole
 triton==2.0.0
     # via torch
-typer==0.7.0
+typer==0.9.0
     # via npe2
-typing-extensions==4.5.0
+typing-extensions==4.6.3
     # via
     #   app-model
     #   magicgui
     #   napari (setup.cfg)
+    #   pint
     #   psygnal
     #   pydantic
     #   superqt
     #   torch
+    #   typer
 tzdata==2023.3
     # via pandas
-urllib3==1.26.15
+urllib3==2.0.3
     # via requests
-virtualenv==20.22.0
+virtualenv==20.23.1
     # via napari (setup.cfg)
 vispy==0.12.2
     # via
@@ -517,15 +529,17 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.4.2
+xarray==2023.5.0
     # via napari (setup.cfg)
-zarr==2.14.2
+zarr==2.15.0
     # via napari (setup.cfg)
 zipp==3.15.0
     # via importlib-metadata
 
 # The following packages are considered to be unsafe in a requirements file:
-setuptools==67.7.2
+pip==23.1.2
+    # via napari-plugin-manager
+setuptools==68.0.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.11.txt b/resources/constraints/constraints_py3.11.txt
index 91f8fbff..d3a85530 100644
--- a/resources/constraints/constraints_py3.11.txt
+++ b/resources/constraints/constraints_py3.11.txt
@@ -42,17 +42,17 @@ click==8.1.3
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.3
+cmake==3.26.4
     # via triton
 comm==0.1.3
     # via ipykernel
-contourpy==1.0.7
+contourpy==1.1.0
     # via matplotlib
-coverage==7.2.5
+coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.5.0
+dask==2023.6.0
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
@@ -70,16 +70,16 @@ executing==1.2.0
     # via stack-data
 fasteners==0.18
     # via zarr
-filelock==3.12.0
+filelock==3.12.2
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.39.4
+fonttools==4.40.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.5.0
+fsspec==2023.6.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -87,28 +87,28 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.75.3
+hypothesis==6.79.1
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.28.1
+imageio==2.31.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
-importlib-metadata==6.6.0
+importlib-metadata==6.7.0
     # via dask
 in-n-out==0.1.7
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.23.1
+ipykernel==6.23.2
     # via
     #   napari-console
     #   qtconsole
-ipython==8.13.2
+ipython==8.14.0
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -128,7 +128,7 @@ jupyter-client==8.2.0
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.0
+jupyter-core==5.3.1
     # via
     #   ipykernel
     #   jupyter-client
@@ -141,7 +141,7 @@ lazy-loader==0.2
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.3
+lit==16.0.6
     # via triton
 locket==1.0.0
     # via partd
@@ -149,9 +149,9 @@ lxml==4.9.2
     # via napari (setup.cfg)
 magicgui==0.7.2
     # via napari (setup.cfg)
-markdown-it-py==2.2.0
+markdown-it-py==3.0.0
     # via rich
-markupsafe==2.1.2
+markupsafe==2.1.3
     # via jinja2
 matplotlib==3.7.1
     # via napari (setup.cfg)
@@ -172,7 +172,9 @@ napari-plugin-engine==0.2.0
     #   napari (setup.cfg)
     #   napari-console
     #   napari-svg
-napari-svg==0.1.6
+napari-plugin-manager==0.1.0a2
+    # via napari (setup.cfg)
+napari-svg==0.1.10
     # via napari (setup.cfg)
 nest-asyncio==1.5.6
     # via ipykernel
@@ -181,10 +183,12 @@ networkx==3.1
     #   scikit-image
     #   torch
 npe2==0.7.0
-    # via napari (setup.cfg)
+    # via
+    #   napari (setup.cfg)
+    #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
-numpy==1.24.3
+numpy==1.25.0
     # via
     #   contourpy
     #   dask
@@ -244,7 +248,7 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.1 ; python_version >= "3.9"
+pandas==2.0.2 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   xarray
@@ -262,14 +266,14 @@ pillow==9.5.0
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.21
+pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.5.1
+platformdirs==3.7.0
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.0.0
+pluggy==1.2.0
     # via pytest
 pooch==1.7.0
     # via
@@ -281,7 +285,7 @@ psutil==5.9.5
     # via
     #   ipykernel
     #   napari (setup.cfg)
-psygnal==0.9.0
+psygnal==0.9.1
     # via
     #   app-model
     #   magicgui
@@ -291,7 +295,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.7
+pydantic==1.10.9
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -305,8 +309,10 @@ pygments==2.15.1
     #   sphinx
     #   superqt
 pyopengl==3.1.6
-    # via napari (setup.cfg)
-pyparsing==3.0.9
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+pyparsing==3.1.0
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
@@ -316,9 +322,9 @@ pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.1
     # via pyqt5
-pyqt6==6.5.0
+pyqt6==6.5.1
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.0
+pyqt6-qt6==6.5.1
     # via pyqt6
 pyqt6-sip==13.5.1
     # via pyqt6
@@ -336,7 +342,7 @@ pyside6-essentials==6.4.2
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.3.1
+pytest==7.3.2
     # via
     #   napari (setup.cfg)
     #   pytest-cov
@@ -344,13 +350,13 @@ pytest==7.3.1
     #   pytest-metadata
     #   pytest-pretty
     #   pytest-qt
-pytest-cov==4.0.0
+pytest-cov==4.1.0
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
 pytest-json-report==1.5.0
     # via -r resources/constraints/version_denylist.txt
-pytest-metadata==2.0.4
+pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
     # via napari (setup.cfg)
@@ -372,7 +378,7 @@ pyyaml==6.0
     #   dask
     #   napari (setup.cfg)
     #   npe2
-pyzmq==25.0.2
+pyzmq==25.1.0
     # via
     #   ipykernel
     #   jupyter-client
@@ -386,18 +392,19 @@ qtpy==2.3.1
     #   magicgui
     #   napari (setup.cfg)
     #   napari-console
+    #   napari-plugin-manager
     #   qtconsole
     #   superqt
-requests==2.30.0
+requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.3.5
+rich==13.4.2
     # via
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-scikit-image==0.20.0
+scikit-image==0.21.0
     # via napari (setup.cfg)
 scipy==1.10.1 ; python_version >= "3.9"
     # via
@@ -440,10 +447,13 @@ superqt==0.4.1
     # via
     #   magicgui
     #   napari (setup.cfg)
+    #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.36
-    # via napari (setup.cfg)
+tensorstore==0.1.37
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
 tifffile==2023.4.12
     # via
     #   napari (setup.cfg)
@@ -476,11 +486,12 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.5.0
+typing-extensions==4.6.3
     # via
     #   app-model
     #   magicgui
     #   napari (setup.cfg)
+    #   pint
     #   psygnal
     #   pydantic
     #   superqt
@@ -488,9 +499,9 @@ typing-extensions==4.5.0
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.2
+urllib3==2.0.3
     # via requests
-virtualenv==20.23.0
+virtualenv==20.23.1
     # via napari (setup.cfg)
 vispy==0.12.2
     # via
@@ -508,15 +519,17 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.4.2
+xarray==2023.5.0
     # via napari (setup.cfg)
-zarr==2.14.2
+zarr==2.15.0
     # via napari (setup.cfg)
 zipp==3.15.0
     # via importlib-metadata
 
 # The following packages are considered to be unsafe in a requirements file:
-setuptools==67.7.2
+pip==23.1.2
+    # via napari-plugin-manager
+setuptools==68.0.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.8.txt b/resources/constraints/constraints_py3.8.txt
index 1ca27026..32c7ea12 100644
--- a/resources/constraints/constraints_py3.8.txt
+++ b/resources/constraints/constraints_py3.8.txt
@@ -30,7 +30,7 @@ build==0.10.0
     # via npe2
 cachey==0.2.1
     # via napari (setup.cfg)
-certifi==2022.12.7
+certifi==2023.5.7
     # via
     #   napari (setup.cfg)
     #   requests
@@ -42,17 +42,17 @@ click==8.1.3
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.3
+cmake==3.26.4
     # via triton
 comm==0.1.3
     # via ipykernel
-contourpy==1.0.7
+contourpy==1.1.0
     # via matplotlib
-coverage==7.2.3
+coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.4.0
+dask==2023.5.0
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
@@ -74,16 +74,16 @@ executing==1.2.0
     # via stack-data
 fasteners==0.18
     # via zarr
-filelock==3.12.0
+filelock==3.12.2
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.39.3
+fonttools==4.40.0
     # via matplotlib
-freetype-py==2.3.0
+freetype-py==2.4.0
     # via vispy
-fsspec==2023.4.0
+fsspec==2023.6.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -91,18 +91,18 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.74.0
+hypothesis==6.79.1
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.28.0
+imageio==2.31.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
-importlib-metadata==6.6.0
+importlib-metadata==6.7.0
     # via
     #   dask
     #   jupyter-client
@@ -115,11 +115,11 @@ in-n-out==0.1.7
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.22.0
+ipykernel==6.23.2
     # via
     #   napari-console
     #   qtconsole
-ipython==8.12.0
+ipython==8.12.2
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -139,7 +139,7 @@ jupyter-client==8.2.0
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.0
+jupyter-core==5.3.1
     # via
     #   ipykernel
     #   jupyter-client
@@ -152,7 +152,7 @@ lazy-loader==0.2
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.2
+lit==16.0.6
     # via triton
 locket==1.0.0
     # via partd
@@ -160,9 +160,9 @@ lxml==4.9.2
     # via napari (setup.cfg)
 magicgui==0.7.2
     # via napari (setup.cfg)
-markdown-it-py==2.2.0
+markdown-it-py==3.0.0
     # via rich
-markupsafe==2.1.2
+markupsafe==2.1.3
     # via jinja2
 matplotlib==3.7.1
     # via napari (setup.cfg)
@@ -183,7 +183,9 @@ napari-plugin-engine==0.2.0
     #   napari (setup.cfg)
     #   napari-console
     #   napari-svg
-napari-svg==0.1.6
+napari-plugin-manager==0.1.0a2
+    # via napari (setup.cfg)
+napari-svg==0.1.10
     # via napari (setup.cfg)
 nest-asyncio==1.5.6
     # via ipykernel
@@ -192,7 +194,9 @@ networkx==3.1
     #   scikit-image
     #   torch
 npe2==0.7.0
-    # via napari (setup.cfg)
+    # via
+    #   napari (setup.cfg)
+    #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
 numpy==1.24.3
@@ -255,7 +259,7 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.1 ; python_version < "3.9"
+pandas==2.0.2 ; python_version < "3.9"
     # via
     #   napari (setup.cfg)
     #   xarray
@@ -273,16 +277,16 @@ pillow==9.5.0
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.20.1
+pint==0.21.1
     # via napari (setup.cfg)
 pkgutil-resolve-name==1.3.10
     # via jsonschema
-platformdirs==3.3.0
+platformdirs==3.7.0
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.0.0
+pluggy==1.2.0
     # via pytest
 pooch==1.7.0
     # via
@@ -294,7 +298,7 @@ psutil==5.9.5
     # via
     #   ipykernel
     #   napari (setup.cfg)
-psygnal==0.9.0
+psygnal==0.9.1
     # via
     #   app-model
     #   magicgui
@@ -304,7 +308,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.7
+pydantic==1.10.9
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -318,8 +322,10 @@ pygments==2.15.1
     #   sphinx
     #   superqt
 pyopengl==3.1.6
-    # via napari (setup.cfg)
-pyparsing==3.0.9
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+pyparsing==3.1.0
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
@@ -329,9 +335,9 @@ pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.1
     # via pyqt5
-pyqt6==6.5.0
+pyqt6==6.5.1
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.0
+pyqt6-qt6==6.5.1
     # via pyqt6
 pyqt6-sip==13.5.1
     # via pyqt6
@@ -349,7 +355,7 @@ pyside6-essentials==6.3.1
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.3.1
+pytest==7.3.2
     # via
     #   napari (setup.cfg)
     #   pytest-cov
@@ -357,13 +363,13 @@ pytest==7.3.1
     #   pytest-metadata
     #   pytest-pretty
     #   pytest-qt
-pytest-cov==4.0.0
+pytest-cov==4.1.0
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
 pytest-json-report==1.5.0
     # via -r resources/constraints/version_denylist.txt
-pytest-metadata==2.0.4
+pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
     # via napari (setup.cfg)
@@ -387,12 +393,12 @@ pyyaml==6.0
     #   dask
     #   napari (setup.cfg)
     #   npe2
-pyzmq==25.0.2
+pyzmq==25.1.0
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.1
+qtconsole==5.4.3
     # via
     #   napari (setup.cfg)
     #   napari-console
@@ -401,20 +407,21 @@ qtpy==2.3.1
     #   magicgui
     #   napari (setup.cfg)
     #   napari-console
+    #   napari-plugin-manager
     #   qtconsole
     #   superqt
-requests==2.28.2
+requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.3.4
+rich==13.4.2
     # via
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-scikit-image==0.20.0
+scikit-image==0.21.0
     # via napari (setup.cfg)
-scipy==1.9.1 ; python_version < "3.9"
+scipy==1.10.1 ; python_version < "3.9"
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -455,10 +462,13 @@ superqt==0.4.1
     # via
     #   magicgui
     #   napari (setup.cfg)
-sympy==1.11.1
+    #   napari-plugin-manager
+sympy==1.12
     # via torch
-tensorstore==0.1.36
-    # via napari (setup.cfg)
+tensorstore==0.1.37
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
 tifffile==2023.4.12
     # via
     #   napari (setup.cfg)
@@ -474,11 +484,11 @@ toolz==0.12.0
     #   dask
     #   napari (setup.cfg)
     #   partd
-torch==2.0.0
+torch==2.0.1
     # via
     #   napari (setup.cfg)
     #   triton
-tornado==6.3.1
+tornado==6.3.2
     # via
     #   ipykernel
     #   jupyter-client
@@ -495,9 +505,9 @@ traitlets==5.9.0
     #   qtconsole
 triton==2.0.0
     # via torch
-typer==0.7.0
+typer==0.9.0
     # via npe2
-typing-extensions==4.5.0
+typing-extensions==4.6.3
     # via
     #   app-model
     #   ipython
@@ -508,11 +518,12 @@ typing-extensions==4.5.0
     #   rich
     #   superqt
     #   torch
+    #   typer
 tzdata==2023.3
     # via pandas
-urllib3==1.26.15
+urllib3==2.0.3
     # via requests
-virtualenv==20.22.0
+virtualenv==20.23.1
     # via napari (setup.cfg)
 vispy==0.12.2
     # via
@@ -532,7 +543,7 @@ wrapt==1.15.0
     # via napari (setup.cfg)
 xarray==2023.1.0
     # via napari (setup.cfg)
-zarr==2.14.2
+zarr==2.15.0
     # via napari (setup.cfg)
 zipp==3.15.0
     # via
@@ -540,7 +551,9 @@ zipp==3.15.0
     #   importlib-resources
 
 # The following packages are considered to be unsafe in a requirements file:
-setuptools==67.7.2
+pip==23.1.2
+    # via napari-plugin-manager
+setuptools==68.0.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.9.txt b/resources/constraints/constraints_py3.9.txt
index bff160ee..a2521a3a 100644
--- a/resources/constraints/constraints_py3.9.txt
+++ b/resources/constraints/constraints_py3.9.txt
@@ -30,7 +30,7 @@ build==0.10.0
     # via npe2
 cachey==0.2.1
     # via napari (setup.cfg)
-certifi==2022.12.7
+certifi==2023.5.7
     # via
     #   napari (setup.cfg)
     #   requests
@@ -42,17 +42,17 @@ click==8.1.3
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.3
+cmake==3.26.4
     # via triton
 comm==0.1.3
     # via ipykernel
-contourpy==1.0.7
+contourpy==1.1.0
     # via matplotlib
-coverage==7.2.3
+coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.4.0
+dask==2023.6.0
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
@@ -74,16 +74,16 @@ executing==1.2.0
     # via stack-data
 fasteners==0.18
     # via zarr
-filelock==3.12.0
+filelock==3.12.2
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.39.3
+fonttools==4.40.0
     # via matplotlib
-freetype-py==2.3.0
+freetype-py==2.4.0
     # via vispy
-fsspec==2023.4.0
+fsspec==2023.6.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -91,18 +91,18 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.74.0
+hypothesis==6.79.1
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.28.0
+imageio==2.31.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
-importlib-metadata==6.6.0
+importlib-metadata==6.7.0
     # via
     #   dask
     #   jupyter-client
@@ -113,11 +113,11 @@ in-n-out==0.1.7
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.22.0
+ipykernel==6.23.2
     # via
     #   napari-console
     #   qtconsole
-ipython==8.12.0
+ipython==8.14.0
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -137,7 +137,7 @@ jupyter-client==8.2.0
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.0
+jupyter-core==5.3.1
     # via
     #   ipykernel
     #   jupyter-client
@@ -150,7 +150,7 @@ lazy-loader==0.2
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.2
+lit==16.0.6
     # via triton
 locket==1.0.0
     # via partd
@@ -158,9 +158,9 @@ lxml==4.9.2
     # via napari (setup.cfg)
 magicgui==0.7.2
     # via napari (setup.cfg)
-markdown-it-py==2.2.0
+markdown-it-py==3.0.0
     # via rich
-markupsafe==2.1.2
+markupsafe==2.1.3
     # via jinja2
 matplotlib==3.7.1
     # via napari (setup.cfg)
@@ -181,7 +181,9 @@ napari-plugin-engine==0.2.0
     #   napari (setup.cfg)
     #   napari-console
     #   napari-svg
-napari-svg==0.1.6
+napari-plugin-manager==0.1.0a2
+    # via napari (setup.cfg)
+napari-svg==0.1.10
     # via napari (setup.cfg)
 nest-asyncio==1.5.6
     # via ipykernel
@@ -190,10 +192,12 @@ networkx==3.1
     #   scikit-image
     #   torch
 npe2==0.7.0
-    # via napari (setup.cfg)
+    # via
+    #   napari (setup.cfg)
+    #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
-numpy==1.24.3
+numpy==1.25.0
     # via
     #   contourpy
     #   dask
@@ -253,7 +257,7 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.1 ; python_version >= "3.9"
+pandas==2.0.2 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   xarray
@@ -271,14 +275,14 @@ pillow==9.5.0
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.20.1
+pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.3.0
+platformdirs==3.7.0
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.0.0
+pluggy==1.2.0
     # via pytest
 pooch==1.7.0
     # via
@@ -290,7 +294,7 @@ psutil==5.9.5
     # via
     #   ipykernel
     #   napari (setup.cfg)
-psygnal==0.9.0
+psygnal==0.9.1
     # via
     #   app-model
     #   magicgui
@@ -300,7 +304,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.7
+pydantic==1.10.9
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -314,8 +318,10 @@ pygments==2.15.1
     #   sphinx
     #   superqt
 pyopengl==3.1.6
-    # via napari (setup.cfg)
-pyparsing==3.0.9
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+pyparsing==3.1.0
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
@@ -325,9 +331,9 @@ pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.1
     # via pyqt5
-pyqt6==6.5.0
+pyqt6==6.5.1
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.0
+pyqt6-qt6==6.5.1
     # via pyqt6
 pyqt6-sip==13.5.1
     # via pyqt6
@@ -345,7 +351,7 @@ pyside6-essentials==6.3.1
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.3.1
+pytest==7.3.2
     # via
     #   napari (setup.cfg)
     #   pytest-cov
@@ -353,13 +359,13 @@ pytest==7.3.1
     #   pytest-metadata
     #   pytest-pretty
     #   pytest-qt
-pytest-cov==4.0.0
+pytest-cov==4.1.0
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
 pytest-json-report==1.5.0
     # via -r resources/constraints/version_denylist.txt
-pytest-metadata==2.0.4
+pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
     # via napari (setup.cfg)
@@ -381,12 +387,12 @@ pyyaml==6.0
     #   dask
     #   napari (setup.cfg)
     #   npe2
-pyzmq==25.0.2
+pyzmq==25.1.0
     # via
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.1
+qtconsole==5.4.3
     # via
     #   napari (setup.cfg)
     #   napari-console
@@ -395,20 +401,21 @@ qtpy==2.3.1
     #   magicgui
     #   napari (setup.cfg)
     #   napari-console
+    #   napari-plugin-manager
     #   qtconsole
     #   superqt
-requests==2.28.2
+requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.3.4
+rich==13.4.2
     # via
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-scikit-image==0.20.0
+scikit-image==0.21.0
     # via napari (setup.cfg)
-scipy==1.9.1 ; python_version >= "3.9"
+scipy==1.10.1 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -449,10 +456,13 @@ superqt==0.4.1
     # via
     #   magicgui
     #   napari (setup.cfg)
-sympy==1.11.1
+    #   napari-plugin-manager
+sympy==1.12
     # via torch
-tensorstore==0.1.36
-    # via napari (setup.cfg)
+tensorstore==0.1.37
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
 tifffile==2023.4.12
     # via
     #   napari (setup.cfg)
@@ -468,11 +478,11 @@ toolz==0.12.0
     #   dask
     #   napari (setup.cfg)
     #   partd
-torch==2.0.0
+torch==2.0.1
     # via
     #   napari (setup.cfg)
     #   triton
-tornado==6.3.1
+tornado==6.3.2
     # via
     #   ipykernel
     #   jupyter-client
@@ -489,23 +499,25 @@ traitlets==5.9.0
     #   qtconsole
 triton==2.0.0
     # via torch
-typer==0.7.0
+typer==0.9.0
     # via npe2
-typing-extensions==4.5.0
+typing-extensions==4.6.3
     # via
     #   app-model
     #   ipython
     #   magicgui
     #   napari (setup.cfg)
+    #   pint
     #   psygnal
     #   pydantic
     #   superqt
     #   torch
+    #   typer
 tzdata==2023.3
     # via pandas
-urllib3==1.26.15
+urllib3==2.0.3
     # via requests
-virtualenv==20.22.0
+virtualenv==20.23.1
     # via napari (setup.cfg)
 vispy==0.12.2
     # via
@@ -523,9 +535,9 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.4.2
+xarray==2023.5.0
     # via napari (setup.cfg)
-zarr==2.14.2
+zarr==2.15.0
     # via napari (setup.cfg)
 zipp==3.15.0
     # via
@@ -533,7 +545,9 @@ zipp==3.15.0
     #   importlib-resources
 
 # The following packages are considered to be unsafe in a requirements file:
-setuptools==67.7.2
+pip==23.1.2
+    # via napari-plugin-manager
+setuptools==68.0.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.9_examples.txt b/resources/constraints/constraints_py3.9_examples.txt
index e69de29b..0d2312a9 100644
--- a/resources/constraints/constraints_py3.9_examples.txt
+++ b/resources/constraints/constraints_py3.9_examples.txt
@@ -0,0 +1,582 @@
+#
+# This file is autogenerated by pip-compile with Python 3.9
+# by the following command:
+#
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.9_examples.txt --resolver=backtracking --strip-extras resources/constraints/version_denylist.txt resources/constraints/version_denylist_examples.txt setup.cfg
+#
+alabaster==0.7.13
+    # via sphinx
+app-model==0.1.4
+    # via napari (setup.cfg)
+appdirs==1.4.4
+    # via
+    #   napari (setup.cfg)
+    #   npe2
+asciitree==0.3.3
+    # via zarr
+asttokens==2.2.1
+    # via stack-data
+attrs==23.1.0
+    # via
+    #   hypothesis
+    #   jsonschema
+babel==2.12.1
+    # via
+    #   napari (setup.cfg)
+    #   sphinx
+backcall==0.2.0
+    # via ipython
+build==0.10.0
+    # via npe2
+cachey==0.2.1
+    # via napari (setup.cfg)
+certifi==2023.5.7
+    # via
+    #   napari (setup.cfg)
+    #   requests
+charset-normalizer==3.1.0
+    # via requests
+click==8.1.3
+    # via
+    #   dask
+    #   typer
+cloudpickle==2.2.1
+    # via dask
+cmake==3.26.4
+    # via triton
+comm==0.1.3
+    # via ipykernel
+contourpy==1.1.0
+    # via matplotlib
+coverage==7.2.7
+    # via pytest-cov
+cycler==0.11.0
+    # via matplotlib
+dask==2023.6.0
+    # via napari (setup.cfg)
+debugpy==1.6.7
+    # via ipykernel
+decorator==5.1.1
+    # via ipython
+distlib==0.3.6
+    # via virtualenv
+docstring-parser==0.15
+    # via magicgui
+docutils==0.17.1
+    # via sphinx
+entrypoints==0.4
+    # via numcodecs
+exceptiongroup==1.1.1
+    # via
+    #   hypothesis
+    #   pytest
+executing==1.2.0
+    # via stack-data
+fasteners==0.18
+    # via zarr
+filelock==3.12.2
+    # via
+    #   torch
+    #   triton
+    #   virtualenv
+fonttools==4.40.0
+    # via matplotlib
+freetype-py==2.4.0
+    # via vispy
+fsspec==2023.6.0
+    # via
+    #   dask
+    #   napari (setup.cfg)
+heapdict==1.0.1
+    # via cachey
+hsluv==5.0.3
+    # via vispy
+hypothesis==6.79.1
+    # via napari (setup.cfg)
+idna==3.4
+    # via requests
+imageio==2.31.1
+    # via
+    #   napari (setup.cfg)
+    #   napari-svg
+    #   scikit-image
+imagesize==1.4.1
+    # via sphinx
+importlib-metadata==6.7.0
+    # via
+    #   dask
+    #   jupyter-client
+    #   sphinx
+importlib-resources==5.12.0
+    # via matplotlib
+in-n-out==0.1.7
+    # via app-model
+iniconfig==2.0.0
+    # via pytest
+ipykernel==6.23.2
+    # via
+    #   napari-console
+    #   qtconsole
+ipython==8.14.0
+    # via
+    #   ipykernel
+    #   napari (setup.cfg)
+    #   napari-console
+ipython-genutils==0.2.0
+    # via qtconsole
+jedi==0.18.2
+    # via ipython
+jinja2==3.1.2
+    # via
+    #   numpydoc
+    #   sphinx
+    #   torch
+joblib==1.2.0
+    # via
+    #   nilearn
+    #   scikit-learn
+jsonschema==4.17.3
+    # via napari (setup.cfg)
+jupyter-client==8.2.0
+    # via
+    #   ipykernel
+    #   qtconsole
+jupyter-core==5.3.1
+    # via
+    #   ipykernel
+    #   jupyter-client
+    #   qtconsole
+kiwisolver==1.4.4
+    # via
+    #   matplotlib
+    #   vispy
+lazy-loader==0.2
+    # via
+    #   napari (setup.cfg)
+    #   scikit-image
+lit==16.0.6
+    # via triton
+locket==1.0.0
+    # via partd
+lxml==4.9.2
+    # via
+    #   napari (setup.cfg)
+    #   nilearn
+magicgui==0.7.2
+    # via napari (setup.cfg)
+markdown-it-py==3.0.0
+    # via rich
+markupsafe==2.1.3
+    # via jinja2
+matplotlib==3.7.1
+    # via napari (setup.cfg)
+matplotlib-inline==0.1.6
+    # via
+    #   ipykernel
+    #   ipython
+mdurl==0.1.2
+    # via markdown-it-py
+mpmath==1.3.0
+    # via sympy
+mypy-extensions==1.0.0
+    # via psygnal
+napari-console==0.0.8
+    # via napari (setup.cfg)
+napari-plugin-engine==0.2.0
+    # via
+    #   napari (setup.cfg)
+    #   napari-console
+    #   napari-svg
+napari-plugin-manager==0.1.0a2
+    # via napari (setup.cfg)
+napari-svg==0.1.10
+    # via napari (setup.cfg)
+nest-asyncio==1.5.6
+    # via ipykernel
+networkx==3.1
+    # via
+    #   scikit-image
+    #   torch
+nibabel==5.1.0
+    # via nilearn
+nilearn==0.10.1
+    # via -r resources/constraints/version_denylist_examples.txt
+npe2==0.7.0
+    # via
+    #   napari (setup.cfg)
+    #   napari-plugin-manager
+numcodecs==0.11.0
+    # via zarr
+numpy==1.23.5
+    # via
+    #   -r resources/constraints/version_denylist_examples.txt
+    #   contourpy
+    #   dask
+    #   imageio
+    #   matplotlib
+    #   napari (setup.cfg)
+    #   napari-svg
+    #   nibabel
+    #   nilearn
+    #   numcodecs
+    #   pandas
+    #   pywavelets
+    #   scikit-image
+    #   scikit-learn
+    #   scipy
+    #   tensorstore
+    #   tifffile
+    #   vispy
+    #   xarray
+    #   zarr
+numpydoc==1.5.0
+    # via napari (setup.cfg)
+nvidia-cublas-cu11==11.10.3.66
+    # via
+    #   nvidia-cudnn-cu11
+    #   nvidia-cusolver-cu11
+    #   torch
+nvidia-cuda-cupti-cu11==11.7.101
+    # via torch
+nvidia-cuda-nvrtc-cu11==11.7.99
+    # via torch
+nvidia-cuda-runtime-cu11==11.7.99
+    # via torch
+nvidia-cudnn-cu11==8.5.0.96
+    # via torch
+nvidia-cufft-cu11==10.9.0.58
+    # via torch
+nvidia-curand-cu11==10.2.10.91
+    # via torch
+nvidia-cusolver-cu11==11.4.0.1
+    # via torch
+nvidia-cusparse-cu11==11.7.4.91
+    # via torch
+nvidia-nccl-cu11==2.14.3
+    # via torch
+nvidia-nvtx-cu11==11.7.91
+    # via torch
+packaging==23.1
+    # via
+    #   -r resources/constraints/version_denylist_examples.txt
+    #   build
+    #   dask
+    #   ipykernel
+    #   matplotlib
+    #   nibabel
+    #   nilearn
+    #   pooch
+    #   pytest
+    #   qtconsole
+    #   qtpy
+    #   scikit-image
+    #   sphinx
+    #   superqt
+    #   vispy
+    #   xarray
+pandas==2.0.2 ; python_version >= "3.9"
+    # via
+    #   napari (setup.cfg)
+    #   nilearn
+    #   xarray
+parso==0.8.3
+    # via jedi
+partd==1.4.0
+    # via dask
+pexpect==4.8.0
+    # via ipython
+pickleshare==0.7.5
+    # via ipython
+pillow==9.5.0
+    # via
+    #   imageio
+    #   matplotlib
+    #   napari (setup.cfg)
+    #   scikit-image
+pint==0.22
+    # via napari (setup.cfg)
+platformdirs==3.7.0
+    # via
+    #   jupyter-core
+    #   pooch
+    #   virtualenv
+pluggy==1.2.0
+    # via pytest
+pooch==1.7.0
+    # via
+    #   napari (setup.cfg)
+    #   scikit-image
+prompt-toolkit==3.0.38
+    # via ipython
+psutil==5.9.5
+    # via
+    #   ipykernel
+    #   napari (setup.cfg)
+psygnal==0.9.1
+    # via
+    #   app-model
+    #   magicgui
+    #   napari (setup.cfg)
+    #   npe2
+ptyprocess==0.7.0
+    # via pexpect
+pure-eval==0.2.2
+    # via stack-data
+pydantic==1.10.9
+    # via
+    #   app-model
+    #   napari (setup.cfg)
+    #   npe2
+pygments==2.15.1
+    # via
+    #   ipython
+    #   napari (setup.cfg)
+    #   qtconsole
+    #   rich
+    #   sphinx
+    #   superqt
+pyopengl==3.1.6
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+pyparsing==3.1.0
+    # via matplotlib
+pyproject-hooks==1.0.0
+    # via build
+pyqt5==5.15.9
+    # via napari (setup.cfg)
+pyqt5-qt5==5.15.2
+    # via pyqt5
+pyqt5-sip==12.12.1
+    # via pyqt5
+pyqt6==6.5.1
+    # via napari (setup.cfg)
+pyqt6-qt6==6.5.1
+    # via pyqt6
+pyqt6-sip==13.5.1
+    # via pyqt6
+pyrsistent==0.19.3
+    # via jsonschema
+pyside2==5.15.2.1 ; python_version != "3.8"
+    # via napari (setup.cfg)
+pyside6==6.3.1 ; python_version < "3.10"
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+pyside6-addons==6.3.1
+    # via pyside6
+pyside6-essentials==6.3.1
+    # via
+    #   pyside6
+    #   pyside6-addons
+pytest==7.3.2
+    # via
+    #   napari (setup.cfg)
+    #   pytest-cov
+    #   pytest-json-report
+    #   pytest-metadata
+    #   pytest-pretty
+    #   pytest-qt
+pytest-cov==4.1.0
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+pytest-json-report==1.5.0
+    # via -r resources/constraints/version_denylist.txt
+pytest-metadata==3.0.0
+    # via pytest-json-report
+pytest-pretty==1.2.0
+    # via napari (setup.cfg)
+pytest-qt==4.2.0
+    # via napari (setup.cfg)
+python-dateutil==2.8.2
+    # via
+    #   jupyter-client
+    #   matplotlib
+    #   pandas
+pytomlpp==1.0.13
+    # via npe2
+pytz==2023.3
+    # via pandas
+pywavelets==1.4.1
+    # via scikit-image
+pyyaml==6.0
+    # via
+    #   dask
+    #   napari (setup.cfg)
+    #   npe2
+pyzmq==25.1.0
+    # via
+    #   ipykernel
+    #   jupyter-client
+    #   qtconsole
+qtconsole==5.4.3
+    # via
+    #   napari (setup.cfg)
+    #   napari-console
+qtpy==2.3.1
+    # via
+    #   magicgui
+    #   napari (setup.cfg)
+    #   napari-console
+    #   napari-plugin-manager
+    #   qtconsole
+    #   superqt
+requests==2.31.0
+    # via
+    #   nilearn
+    #   pooch
+    #   sphinx
+rich==13.4.2
+    # via
+    #   napari (setup.cfg)
+    #   npe2
+    #   pytest-pretty
+scikit-image==0.21.0
+    # via napari (setup.cfg)
+scikit-learn==1.2.2
+    # via nilearn
+scipy==1.10.1 ; python_version >= "3.9"
+    # via
+    #   napari (setup.cfg)
+    #   nilearn
+    #   scikit-image
+    #   scikit-learn
+shiboken2==5.15.2.1
+    # via pyside2
+shiboken6==6.3.1
+    # via
+    #   pyside6
+    #   pyside6-addons
+    #   pyside6-essentials
+six==1.16.0
+    # via
+    #   asttokens
+    #   python-dateutil
+snowballstemmer==2.2.0
+    # via sphinx
+sortedcontainers==2.4.0
+    # via hypothesis
+sphinx==4.5.0
+    # via
+    #   napari (setup.cfg)
+    #   numpydoc
+sphinxcontrib-applehelp==1.0.4
+    # via sphinx
+sphinxcontrib-devhelp==1.0.2
+    # via sphinx
+sphinxcontrib-htmlhelp==2.0.1
+    # via sphinx
+sphinxcontrib-jsmath==1.0.1
+    # via sphinx
+sphinxcontrib-qthelp==1.0.3
+    # via sphinx
+sphinxcontrib-serializinghtml==1.1.5
+    # via sphinx
+stack-data==0.6.2
+    # via ipython
+superqt==0.4.1
+    # via
+    #   magicgui
+    #   napari (setup.cfg)
+    #   napari-plugin-manager
+sympy==1.12
+    # via torch
+tensorstore==0.1.37
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+threadpoolctl==3.1.0
+    # via scikit-learn
+tifffile==2023.4.12
+    # via
+    #   napari (setup.cfg)
+    #   scikit-image
+tomli==2.0.1
+    # via
+    #   build
+    #   coverage
+    #   pyproject-hooks
+    #   pytest
+toolz==0.12.0
+    # via
+    #   dask
+    #   napari (setup.cfg)
+    #   partd
+torch==2.0.1
+    # via
+    #   napari (setup.cfg)
+    #   triton
+tornado==6.3.2
+    # via
+    #   ipykernel
+    #   jupyter-client
+tqdm==4.65.0
+    # via napari (setup.cfg)
+traitlets==5.9.0
+    # via
+    #   comm
+    #   ipykernel
+    #   ipython
+    #   jupyter-client
+    #   jupyter-core
+    #   matplotlib-inline
+    #   qtconsole
+triton==2.0.0
+    # via torch
+typer==0.9.0
+    # via npe2
+typing-extensions==4.6.3
+    # via
+    #   app-model
+    #   ipython
+    #   magicgui
+    #   napari (setup.cfg)
+    #   pint
+    #   psygnal
+    #   pydantic
+    #   superqt
+    #   torch
+    #   typer
+tzdata==2023.3
+    # via pandas
+urllib3==2.0.3
+    # via requests
+virtualenv==20.23.1
+    # via napari (setup.cfg)
+vispy==0.12.2
+    # via
+    #   napari (setup.cfg)
+    #   napari-svg
+wcwidth==0.2.6
+    # via prompt-toolkit
+wheel==0.40.0
+    # via
+    #   nvidia-cublas-cu11
+    #   nvidia-cuda-cupti-cu11
+    #   nvidia-cuda-runtime-cu11
+    #   nvidia-curand-cu11
+    #   nvidia-cusparse-cu11
+    #   nvidia-nvtx-cu11
+wrapt==1.15.0
+    # via napari (setup.cfg)
+xarray==2023.5.0
+    # via napari (setup.cfg)
+zarr==2.15.0
+    # via napari (setup.cfg)
+zipp==3.15.0
+    # via
+    #   importlib-metadata
+    #   importlib-resources
+
+# The following packages are considered to be unsafe in a requirements file:
+pip==23.1.2
+    # via napari-plugin-manager
+setuptools==68.0.0
+    # via
+    #   nvidia-cublas-cu11
+    #   nvidia-cuda-cupti-cu11
+    #   nvidia-cuda-runtime-cu11
+    #   nvidia-curand-cu11
+    #   nvidia-cusparse-cu11
+    #   nvidia-nvtx-cu11
diff --git a/resources/constraints/version_denylist.txt b/resources/constraints/version_denylist.txt
index a195b6f2..629e74f5 100644
--- a/resources/constraints/version_denylist.txt
+++ b/resources/constraints/version_denylist.txt
@@ -1,4 +1,6 @@
 pytest-cov
 PySide6 < 6.3.2 ; python_version < '3.10'
-PySide6 != 6.4.3, !=6.5.0 ; python_version >= '3.10'
+PySide6 != 6.4.3, !=6.5.0, !=6.5.1, !=6.5.1.1; python_version >= '3.10'
 pytest-json-report
+pyopengl!=3.1.7
+tensorstore!=0.1.38
diff --git a/resources/requirements_mypy.txt b/resources/requirements_mypy.txt
index 9eff78eb..88dcaefe 100644
--- a/resources/requirements_mypy.txt
+++ b/resources/requirements_mypy.txt
@@ -1,22 +1,80 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.11
+# by the following command:
 #
-#    pip-compile requirements_mypy.in
+#    pip-compile --output-file=resources/requirements_mypy.txt --resolver=backtracking resources/requirements_mypy.in
 #
-mypy==1.2.0
-    # via -r requirements_mypy.in
+appdirs==1.4.4
+    # via npe2
+build==0.10.0
+    # via npe2
+click==8.1.3
+    # via typer
+docstring-parser==0.15
+    # via magicgui
+magicgui==0.7.2
+    # via -r resources/requirements_mypy.in
+markdown-it-py==3.0.0
+    # via rich
+mdurl==0.1.2
+    # via markdown-it-py
+mypy==1.4.0
+    # via -r resources/requirements_mypy.in
 mypy-extensions==1.0.0
-    # via mypy
-tomli==2.0.1
-    # via mypy
-types-pyyaml==6.0.12.9
-    # via -r requirements_mypy.in
-types-requests==2.28.11.17
-    # via -r requirements_mypy.in
-types-setuptools==67.6.0.7
-    # via -r requirements_mypy.in
-types-urllib3==1.26.25.10
+    # via
+    #   mypy
+    #   psygnal
+npe2==0.7.0
+    # via -r resources/requirements_mypy.in
+numpy==1.25.0
+    # via -r resources/requirements_mypy.in
+packaging==23.1
+    # via
+    #   build
+    #   qtpy
+    #   superqt
+psygnal==0.9.1
+    # via
+    #   magicgui
+    #   npe2
+pydantic==1.10.9
+    # via
+    #   -r resources/requirements_mypy.in
+    #   npe2
+pygments==2.15.1
+    # via
+    #   rich
+    #   superqt
+pyproject-hooks==1.0.0
+    # via build
+pytomlpp==1.0.13
+    # via npe2
+pyyaml==6.0
+    # via npe2
+qtpy==2.3.1
+    # via
+    #   -r resources/requirements_mypy.in
+    #   magicgui
+    #   superqt
+rich==13.4.2
+    # via npe2
+superqt==0.4.1
+    # via magicgui
+typer==0.9.0
+    # via npe2
+types-pyyaml==6.0.12.10
+    # via -r resources/requirements_mypy.in
+types-requests==2.31.0.1
+    # via -r resources/requirements_mypy.in
+types-setuptools==68.0.0.0
+    # via -r resources/requirements_mypy.in
+types-urllib3==1.26.25.13
     # via types-requests
-typing-extensions==4.5.0
-    # via mypy
+typing-extensions==4.6.3
+    # via
+    #   magicgui
+    #   mypy
+    #   psygnal
+    #   pydantic
+    #   superqt
+    #   typer
diff --git a/setup.cfg b/setup.cfg
index 69c42ec9..0bb61b91 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -48,8 +48,9 @@ install_requires =
     cachey>=0.2.1
     certifi>=2018.1.18
     dask[array]>=2.15.0,!=2.28.0  # https://github.com/napari/napari/issues/1656
-    imageio>=2.5.0,!=2.11.0,!=2.22.1
+    imageio>=2.20,!=2.22.1
     jsonschema>=3.2.0
+    lazy_loader>=0.2
     magicgui>=0.3.6
     napari-console>=0.0.6
     napari-plugin-engine>=0.1.9
@@ -72,7 +73,7 @@ install_requires =
     scipy>=1.4.1 ; python_version < '3.9'
     scipy>=1.5.4 ; python_version >= '3.9'
     sphinx>=4.3.0,<5  # numpydoc dependency. sphinx>=5 breaks the docs build; see https://github.com/napari/napari/pull/4915
-    superqt>=0.3.0
+    superqt>=0.4.1
     tifffile>=2020.2.16
     toolz>=0.10.0
     tqdm>=4.56.0
@@ -108,6 +109,7 @@ qt =  # alias for pyqt5
 # all is the full "batteries included" extra.
 all =
     %(pyqt5)s
+    napari-plugin-manager >=0.1.0a1, <0.2.0
 # optional (i.e. opt-in) packages, see https://github.com/napari/napari/pull/3867#discussion_r864354854
 optional =
     triangle
diff --git a/tools/check_updated_packages.py b/tools/check_updated_packages.py
index 0095aed1..1e2326f7 100644
--- a/tools/check_updated_packages.py
+++ b/tools/check_updated_packages.py
@@ -1,10 +1,13 @@
 import argparse
+import os
 import re
 import subprocess  # nosec
 import sys
 from configparser import ConfigParser
 from pathlib import Path
 
+from get_auto_upgrade_branch_name import get_base_branch_name
+
 name_re = re.compile(r"[\w-]+")
 changed_name_re = re.compile(r"\+([\w-]+)")
 
@@ -15,16 +18,55 @@ parser = argparse.ArgumentParser()
 parser.add_argument("--main-packages", action="store_true")
 args = parser.parse_args()
 
+ref_name = os.environ.get("GITHUB_REF_NAME")
+event = os.environ.get("GITHUB_EVENT_NAME", "")
+if ref_name is None:
+    out = subprocess.run(  # nosec
+        ["git", "rev-parse", "--abbrev-ref", "HEAD"],
+        capture_output=True,
+        check=True,
+    )
+    ref_name = out.stdout.decode().strip()
+
+base_branch = get_base_branch_name(ref_name, event)
+
 out = subprocess.run(  # nosec
-    [
-        "git",
-        "diff",
-        str(src_dir / "resources" / "constraints" / "constraints_py3.10.txt"),
-    ],
+    ["git", "branch", "--list", "--format", "%(refname:short)", "-a"],
     capture_output=True,
     check=True,
 )
 
+branches = out.stdout.decode().split("\n")
+
+if base_branch not in branches:
+    if f"origin/{base_branch}" not in branches:
+        print(f"base branch {base_branch} not found in {branches!r}")
+        sys.exit(1)
+
+    base_branch = f"origin/{base_branch}"
+
+
+try:
+    out = subprocess.run(  # nosec
+        [
+            "git",
+            "diff",
+            base_branch,
+            str(
+                src_dir
+                / "resources"
+                / "constraints"
+                / "constraints_py3.10.txt"
+            ),
+        ],
+        capture_output=True,
+        check=True,
+    )
+except subprocess.CalledProcessError as e:
+    print("stderr", e.stderr.decode())
+    print("stdout", e.stdout.decode())
+    sys.exit(1)
+
 changed_packages = [
     changed_name_re.match(x)[1].lower()
     for x in out.stdout.decode().split("\n")
diff --git a/tools/get_auto_upgrade_branch_name.py b/tools/get_auto_upgrade_branch_name.py
new file mode 100644
index 00000000..d2facee5
--- /dev/null
+++ b/tools/get_auto_upgrade_branch_name.py
@@ -0,0 +1,60 @@
+import argparse
+import os
+
+DEFAULT_NAME = "auto-dependency-upgrades"
+
+
+def main():
+    ref_name = os.environ.get("GITHUB_REF_NAME")
+    event = os.environ.get("GITHUB_EVENT_NAME")
+    repository = os.environ.get("GITHUB_REPOSITORY")
+
+    parse = argparse.ArgumentParser()
+    parse.add_argument("--verbose", action="store_true")
+    parse.add_argument("--base_branch", action="store_true")
+    args = parse.parse_args()
+
+    if args.verbose:
+        print(f"ref_name: {ref_name}")
+        print(f"event: {event}")
+        print(f"repository: {repository}")
+        print(f"base branch: {get_base_branch_name(ref_name, event)}")
+        print(f"branch: {get_branch_name(ref_name, event, repository)}")
+    elif args.base_branch:
+        print(get_base_branch_name(ref_name, event))
+    else:
+        print(get_branch_name(ref_name, event, repository))
+
+
+def get_base_branch_name(ref_name, event):
+    if ref_name == DEFAULT_NAME:
+        return "main"
+    if ref_name.startswith(DEFAULT_NAME):
+        if event in {"pull_request", "pull_request_target"}:
+            return os.environ.get("GITHUB_BASE_REF")
+        return ref_name[len(DEFAULT_NAME) + 1 :]
+    return ref_name
+
+
+def get_branch_name(ref_name, event, repository):
+    if event == "schedule":
+        return DEFAULT_NAME
+
+    if event == "workflow_dispatch":
+        if ref_name == "main":
+            return DEFAULT_NAME
+        return f"{DEFAULT_NAME}-{ref_name}"
+
+    if event == "push":
+        if ref_name.startswith(DEFAULT_NAME):
+            return ref_name
+        if ref_name != "main":
+            return f"{DEFAULT_NAME}-{ref_name}"
+    if event == "pull_request" and repository == "napari/napari":
+        return ref_name
+
+    return "skip"
+
+
+if __name__ == "__main__":
+    main()
-- 
2.34.1
